django/utils/http.py

```diff
@@ -175,7 +175,9 @@ def parse_http_date(date):
     try:
         year = int(m.group('year'))
         if year < 100:
-            if year < 70:
+            current_year = datetime.datetime.utcnow().year - 2000  # 2018 -> 18
+            # parsing done in reference to current year as in RFC7231
+            if year <= current_year + 50:
```

Comment:
I think this fails for some cases in the current year as well as for some in the future.

``` python
def get_year(year, current_year):
    assert -100 < year < 100
	if year <= current_year + 50:
		year += 2000
	else:
		year += 1900
	return year

assert get_year(19, current_year=2019) == 2019, "same year"
assert get_year(20, current_year=2019) == 2020, "next year"  
assert get_year(18, current_year=2019) == 2018, "last year"  
assert get_year(40, current_year=2019) == 2040, "21y hence"
assert get_year(80, current_year=2019) == 1980, "39y ago, not 61y hence"  # fails!
assert get_year(10, current_year=2019) == 2010, "9y ago, not 91y hence"  
assert get_year(69, current_year=2019) == 2069, "50y hence"  
assert get_year(80, current_year=2070) == 2080, "future 10y hence"  
assert get_year(60, current_year=2070) == 2060, "future 10y ago, not 90y hence"  
assert get_year(10, current_year=2070) == 2110, "future 40y hence"  # fails!
assert get_year(21, current_year=2070) == 2021, "future 49y ago, not 51y hence"
assert get_year(20, current_year=2070) == 2120, "future 50y hence"  # fails!
```


Can I suggest an alternative implementation?
``` python

def get_year(short_year, current_year):
    assert -100 < year < 100
    current_short_year = current_year % 100
    delta = short_year - current_short_year
    if delta < 0:
    	delta += 100  # assume future
    if delta > 50:
        delta -= 100  # then shorten if too far in the future
    return current_year + delta
```